loader/loader.c

```diff
@@ -5568,18 +5568,18 @@ VkResult loader_validate_instance_extensions(struct loader_instance *inst, const
         // Not in global list, search layer extension lists
         struct loader_layer_properties *layer_prop = NULL;
         for (uint32_t j = 0; NULL == extension_prop && j < expanded_layers.count; ++j) {
-            layer_prop = loaderFindLayerProperty(pCreateInfo->ppEnabledLayerNames[j], instance_layers);
-            if (NULL == layer_prop) {
-                // Should NOT get here, loaderValidateLayers should have already filtered this case out.
-                continue;
-            }
-
             extension_prop =
                 get_extension_property(pCreateInfo->ppEnabledExtensionNames[i], &expanded_layers.list[j].instance_extension_list);
             if (extension_prop) {
                 // Found the extension in one of the layers enabled by the app.
                 break;
             }
+
+            layer_prop = loaderFindLayerProperty(pCreateInfo->ppEnabledLayerNames[j], instance_layers);
+            if (NULL == layer_prop) {
+                // Should NOT get here, loaderValidateLayers should have already filtered this case out.
+                continue;
+            }
```

Comment:
Yeah this actually causes a segfault. If an implicit layer is present (which our WSI layer that we're working on is), then expanded_layers.count is > 0, but the pCreateInfo->enabledLayerCount is 0 and the pCreateInfo->ppEnabledLayerNames is NULL. I'd actually suggest removing the code because, as it states we should never get there in the first place. "Should NOT get here, loaderValidateLayers should have already filtered this case out."

What do you think? Reorder, remove or something else?

Reply:
I'd say leave this as is. I'll look into exactly why this is happening.